### PR TITLE
libc: Implement vscanf() function

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>
Change-Id: I2421e57b2c848c43afb749b8ebfa79ec457cde26

### CODE REVIEW

File: libs/libc/stdio/lib_vprintf.c [hunk 1]
```diff
@@ -49,7 +52,13 @@
...
 int vprintf(FAR const IPTR char *fmt, va_list ap)
 {
+#if CONFIG_NFILE_STREAMS > 0
   /* vfprintf into stdout */
 
   return vfprintf(stdout, fmt, ap);
+#else
+  /* vsyslog into debug channel */
+
+  return nx_vsyslog(LOG_INFO, fmt, &ap);
```
Comment (patacongo) [inline libs/libc/stdio/lib_vprintf.c:62]: It would be much better to use vdprintf()

Comment (patacongo) [reply libs/libc/stdio/lib_vprintf.c:62]: vdprintf() will behave exactly like printf, but will not depend on the FILE stdout.  That is a better choice.

Comment (patacongo) [reply libs/libc/stdio/lib_vprintf.c:62]: syslog() has other properties and may not output of fd == 0 as is REQUIRED for printf().

Comment (patacongo) [inline libs/libc/stdio/lib_vprintf.c:62]:
```suggestion
  return vdprintf(1, fmt, &ap);
```
This is the perfect replacement!

Comment (patacongo) [reply libs/libc/stdio/lib_vprintf.c:62]: If you want that kind of nonstandard behavior from printf for some reason. You probably could create a configuration option to redirect print to syslog.  But I think it is unacceptable as default behavior for printf() because it breaks it required behavior.

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_vprintf.c]: Done.

File: include/stdio.h [hunk 2]
```diff
@@ -192,10 +192,11 @@ int    vfprintf(FAR FILE *stream, FAR const IPTR char *fmt,
          va_list ap);
 int    vfscanf(FAR FILE *stream, FAR const IPTR char *fmt, va_list ap);
 int    vprintf(FAR const IPTR char *fmt, va_list ap);
+int    vscanf(FAR const IPTR char *fmt, va_list ap);
```
Comment (patacongo) [inline include/stdio.h:195]: These changes seem out of place in the PR.  Can we move this to a different PR.  We should not make any IPTR-related changes until we have a clear understanding and a plan for how we want to handle these.

File: libs/libc/stdio/lib_printf.c [hunk 3]
```diff
@@ -59,11 +55,7 @@ int printf(FAR const IPTR char *fmt, ...)
   int     ret;
 
   va_start(ap, fmt);
-#if CONFIG_NFILE_STREAMS > 0
-  ret = vfprintf(stdout, fmt, ap);
-#else
-  ret = nx_vsyslog(LOG_INFO, fmt, &ap);
-#endif
+  ret = vdprintf(STDOUT_FILENO, fmt, ap);
```
Comment (patacongo) [inline libs/libc/stdio/lib_printf.c:58]: I already merged this, but it is not correct.  printf must used streams and buffere C I/O if available.

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_printf.c:58]: Yes, you are right. I forget the I/O buffer.
